src/steps/intune/steps/noncomplianceFindings/converters.ts

```diff
@@ -0,0 +1,127 @@
+import {
+  createDirectRelationship,
+  createIntegrationEntity,
+  parseTimePropertyValue,
+  Relationship,
+} from '@jupiterone/integration-sdk-core';
+import { DeviceConfigurationDeviceStatus } from '@microsoft/microsoft-graph-types-beta';
+import { entities, relationships } from '../../constants';
+import {
+  DeviceConfigurationEntity,
+  ManagedDeviceEntity,
+  NoncomplianceFindingEntity,
+} from '../../types';
+
+// https://docs.microsoft.com/en-us/graph/api/resources/intune-deviceconfig-deviceconfigurationdevicestatus?view=graph-rest-beta
+export function createNoncomplianceFindingEntity(
+  deviceStatus: DeviceConfigurationDeviceStatus,
+  deviceConfigurationEntity: DeviceConfigurationEntity,
+): NoncomplianceFindingEntity {
+  return createIntegrationEntity({
+    entityData: {
+      source: deviceStatus,
+      assign: {
+        _class: entities.NONCOMPLIANCE_FINDING._class,
+        _type: entities.NONCOMPLIANCE_FINDING._type,
+        id: deviceStatus.id,
+        name: 'Latest finding from ' + deviceConfigurationEntity.displayName,
+        category: 'endpoint',
+        assessment: deviceConfigurationEntity.displayName,
+        status: deviceStatus.status, // Possible values are: unknown, notApplicable, compliant, remediated, nonCompliant, error, conflict, notAssigned.
+        severity: calculateSeverity(deviceStatus.status),
+        numericSeverity: calculateNumericSeverity(deviceStatus.status),
+        open: findingIsOpen(deviceStatus.status),
+        lastProcessedOn: parseTimePropertyValue(
+          deviceStatus.lastReportedDateTime,
+        ),
+        lastTestedOn: parseTimePropertyValue(deviceStatus.lastReportedDateTime),
+        lastUpdatedOn: parseTimePropertyValue(
+          deviceStatus.lastReportedDateTime,
+        ),
+      },
+    },
+  }) as NoncomplianceFindingEntity;
+}
+
+export function findingIsOpen(
+  deviceStatus: DeviceConfigurationDeviceStatus['status'],
+) {
+  return ![
+    'notApplicable',
+    'compliant',
+    'remediated',
+    'notAssigned',
+    'unknown',
+    undefined,
+  ].includes(deviceStatus);
+}
```

Comment:
Do you think it'd be worth returning a ternary from this function?

```suggestion
export function findingIsOpen(
  deviceStatus: DeviceConfigurationDeviceStatus['status'],
): true | false | undefined {
  const closedDeviceStatuses = [
    'notApplicable',
    'compliant',
    'remediated',
    'notAssigned',
    'unknown',
  ];
  if (closedDeviceStatuses.includes(deviceStatus) {
    return true;
  }
 
  const openDeviceStatuses = [
    // figure out what signifies that this is open
  ];
  if (openDeviceStatuses.includes(deviceStatus) {
    return false;
  }
  
  if (deviceStatus === undefined) {
    return undefined;
  }
  
  logger.warn({ deviceStatus }, 'Unexpected device status');
  return undefined;
}
```

Reply:
Yeah this would be a bit safer in case they add more types in the future.

- [ ] be more explicit on if findings are open or not.